service/src/main/java/skills/services/admin/moveSkills/SkillsMoveService.groovy

```diff
@@ -0,0 +1,239 @@
+                if (!moveSkillBetweenGroupsInTheSameSubject) {
+                    ruleSetDefGraphService.removeGraphRelationship(projectId, subject.skillId, SkillDef.ContainerType.Subject, projectId, skillId, SkillRelDef.RelationshipType.GroupSkillToSubject)
+                }
+            } else {
+                ruleSetDefGraphService.removeGraphRelationship(projectId, parentSkill.skillId, SkillDef.ContainerType.Subject, projectId, skillId, SkillRelDef.RelationshipType.RuleSetDefinition)
+            }
+
+            // handle destination
+            if (isGroupDest) {
+                ruleSetDefGraphService.assignGraphRelationship(projectId, skillReuseRequest.groupId, SkillDef.ContainerType.SkillsGroup, projectId, skillId, SkillRelDef.RelationshipType.SkillsGroupRequirement)
+                if (!moveSkillBetweenGroupsInTheSameSubject) {
+                    ruleSetDefGraphService.assignGraphRelationship(projectId, skillReuseRequest.subjectId, SkillDef.ContainerType.Subject, projectId, skillId, SkillRelDef.RelationshipType.GroupSkillToSubject)
+                }
+            } else {
+                ruleSetDefGraphService.assignGraphRelationship(projectId, skillReuseRequest.subjectId, SkillDef.ContainerType.Subject, projectId, skillId, SkillRelDef.RelationshipType.RuleSetDefinition)
+            }
+        }
+
+        resetDisplayOrder(origParentSkill, destParentSkillId)
+        return origParentSkill
+    }
+
+    private void resetDisplayOrder(SkillDef origParentSkill, String parentSkillId) {
+        List<SkillDef> siblingsOrig = ruleSetDefGraphService.getChildrenSkills(origParentSkill, [SkillRelDef.RelationshipType.RuleSetDefinition, SkillRelDef.RelationshipType.SkillsGroupRequirement])
+        displayOrderService.resetDisplayOrder(siblingsOrig)
+        SkillDef desParent = skillDefAccessor.getSkillDef(origParentSkill.projectId, parentSkillId, [SkillDef.ContainerType.Subject, SkillDef.ContainerType.SkillsGroup])
+        List<SkillDef> siblingsDest = ruleSetDefGraphService.getChildrenSkills(desParent, [SkillRelDef.RelationshipType.RuleSetDefinition, SkillRelDef.RelationshipType.SkillsGroupRequirement])
+        displayOrderService.resetDisplayOrder(siblingsDest)
+    }
+
+    @Profile
+    private void updateOrigDefinitionPoints(String projectId, SkillDef origParentSkill, SkillDef destSubj) {
+        // update points for the origination group and/or subject
+        // optimization - handle the case where skill was moved from a group to its parent subject
+        if (destSubj.skillId != origParentSkill.skillId) {
+            if (origParentSkill.type == SkillDef.ContainerType.SkillsGroup) {
+                skillCatalogTransactionalAccessor.updateGroupTotalPoints(projectId, origParentSkill.skillId)
+                SkillDef subject = ruleSetDefGraphService.getParentSkill(origParentSkill.id)
```

Comment:
Suggest renaming the SkillsCatalogTransactionAccessor since it appears to be used for non-catalog purposes as well, could be a bit confusing to come across later.

Reply:
:+1: